airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -266,6 +286,27 @@ public JobInfoRead resetConnection(final ConnectionIdRequestBody connectionIdReq
     return JobConverter.getJobInfoRead(job);
   }
 
+  public JobInfoRead cancelJob(JobIdRequestBody jobIdRequestBody) throws IOException {
+    final long jobId = jobIdRequestBody.getId();
+
+    // first prevent this job from being scheduled again
+    jobPersistence.cancelJob(jobId);
```

Comment:
could you add an issue (or update the blocking issue) to mention we should update the hacky part and then link that in a TODO?